src/software/gui/full_system/widgets/ai_control.cpp

```diff
@@ -209,49 +211,83 @@ void setupChannelSpinBox(Ui::AutogeneratedFullSystemMainWidget *widget,
     setupSpinBox(spin_box, channel_parameter);
 }
 
-void setupFriendlyGoalieIDSpinBox(
-    Ui::AutogeneratedFullSystemMainWidget *widget,
-    std::shared_ptr<Parameter<int>> friendly_goalie_id_parameter)
+void setupFriendlyGoalieIDComboBox(
+        Ui::AutogeneratedFullSystemMainWidget *widget,
+        std::shared_ptr<Parameter<bool>> override_friendly_goalie_id_parameter,
+        std::shared_ptr<Parameter<int>> friendly_goalie_id_parameter)
 {
-    auto parameter = friendly_goalie_id_parameter;
-    auto spin_box  = widget->friendly_goalie_id_spin_box;
-
-    spin_box->setValue(parameter->value());
-    spin_box->setMinimum(0);
-    spin_box->setMaximum(std::numeric_limits<uint8_t>::max());
-
-    auto on_channel_changed = [parameter](int new_value) {
-        parameter->setValue(new_value);
-    };
+    setupGoalieIDComboBox(widget, override_friendly_goalie_id_parameter, friendly_goalie_id_parameter, "friendly");
+}
 
-    // QSpinBox has 2 "valueChanged" signals that each provide different info (string vs
-    // int), so we need to static_cast to specify the integer version
-    QWidget::connect(spin_box,
-                     static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
-                     on_channel_changed);
+void setupEnemyGoalieIDComboBox(
+        Ui::AutogeneratedFullSystemMainWidget *widget,
+        std::shared_ptr<Parameter<bool>> override_enemy_goalie_id_parameter,
+        std::shared_ptr<Parameter<int>> enemy_goalie_id_parameter
+        )
+{
+    setupGoalieIDComboBox(widget, override_enemy_goalie_id_parameter, enemy_goalie_id_parameter, "enemy");
 
-    setupSpinBox(spin_box, parameter);
 }
 
-void setupEnemyGoalieIDSpinBox(Ui::AutogeneratedFullSystemMainWidget *widget,
-                               std::shared_ptr<Parameter<int>> enemy_goalie_id_parameter)
+void setupGoalieIDComboBox(
+        Ui::AutogeneratedFullSystemMainWidget *widget,
+        std::shared_ptr<Parameter<bool>> override_goalie_id_parameter,
+        std::shared_ptr<Parameter<int>> goalie_id_parameter,
+        std::string friendly_or_enemy
+        )
```

Comment:
we should pass in a combo box pointer instead so we don't have to differentiate between friendly and enemy. (if we actually did want to pass in a friendly or enemy flag then we'd use enums instead of strings for better type safety)